src/waypoint-hoc.jsx

```diff
@@ -36,6 +40,9 @@ const waypoint = Component => {
     constructor(props) {
       super(props);
       this.state = { scrolled: {} };
+      this._handleScroll = (this.props.throttle ?
+        throttle(this._handleScroll, this.props.throttle) :
+        this._handleScroll).bind(this);

```

Comment:
What merited the throttle? I understand that it can improve performance, but I'd like to avoid introducing this complexity unless we are certain that we need it. The commit message needs to explain this. And I want this to be in a separate PR, complete with tests, a good commit message and possibly a benchmark script.